### PR TITLE
Wagtail 3.0 Support

### PR DESCRIPTION
**Pull request by BrianXu20**
[Ticket
](https://projects.torchbox.com/projects/support-team/tickets/491)

Make the current package support Wagtail 3.0. 

- Drop Django 2.2 (EOL)
- Drop Python 3.6 (EOL)
- Drop Wagtail <2.15 (EOL)
- Add Wagtail 3.0 support

### CODE REVIEW

**Review by nickmoreton — CHANGES_REQUESTED** [review]
A couple of small suggestions.

File: setup.cfg [hunk 1]
```diff
@@ -14,12 +14,12 @@ license_files =
...
+python_requires = >= 3.7
 setup_requires =
   setuptools >= 40.6
   pip >= 10
 include_package_data = true
 packages = find:
 install_requires =
-    Django>=2.2, <3.3
-    wagtail>=2.5, <3
+    Django>=3.2, <5.0
```
[the remark this answers is not in the document]
Comment (BrianXu20) [reply setup.cfg]: I think Wagtail 3.0 supports Django4.0, and you can't install wagtail <3.0 with Django4.0 that's why I did this. However, I think we can limit it to Django3 as the priority is to support Wagtail3.0

**Review by nickmoreton — APPROVED** [review]
LGTM! Thanks Brian

**Comment by BrianXu20** [comment]
Hi @zerolab , 
Can we have this merge in and have a release? 
Thank you

**Review by kevinhowbrook — APPROVED** [review]
Thanks @BrianXu20 I've updated this with master and bumped the version for a new release

**Comment by jsma** [comment]
I'm not sure I understand the reasoning and decision above https://github.com/torchbox/wagtail-footnotes/pull/29#discussion_r891258536

If someone is using this package with a project that is pinned to Wagtail 2.15, Wagtail itself will constrain the version of Django installed (assuming the project hasn't also pinned the version of Django). But by restricting Django to <4.0 here, this prevents projects from running Wagtail 3 & Django 4.0.x when using wagtail-footnotes.

E.g. with requirements.txt:
```
wagtail==3.0.1
django==4.0.5
# Install from Github since 0.8.0 is not on PyPI yet
-e git+https://github.com/torchbox/wagtail-footnotes@0.8.0#egg=wagtail-footnotes
```

```
pip install -r requirements.txt
...snip...
ERROR: Cannot install -r requirements.txt (line 1), -r requirements.txt (line 3) and django==4.0.5 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested django==4.0.5
    wagtail 3.0.1 depends on Django<4.1 and >=3.2
    wagtail-footnotes 0.8.0 depends on Django<4.0 and >=3.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

**Pull request merged**